### PR TITLE
Remove read permission for jks when ssl is enabled

### PR DESCRIPTION
**Pull request by sajinidesilva**

### CODE REVIEW

File: modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQConnection.java [hunk 1]
```diff
@@ -610,16 +622,75 @@ public boolean attemptReconnection()
...
+            throw (E) e.getException();
+        } finally {
+            return (T) returnObject;
+        }
+    }
+
+
+
+
+//    public <T, E extends Exception> T executeRetrySupport(final FailoverProtectedOperation<T,E> operation) throws E
```
Comment (pumudu88) [inline modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQConnection.java]: do we need this commented codes ?

File: modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQConnection.java [hunk 2]
```diff
@@ -880,8 +951,23 @@ private void doClose(List<AMQSession> sessions, long timeout) throws JMSExceptio
...
+                        // Requires permission java.lang.RuntimePermission "modifyThread"
+                        AccessController.doPrivileged(new PrivilegedAction<Void>( ){
+                            @Override
+                            public Void run() {
+                                 _taskPool.shutdown();
+                                return null;
+                            }
+                        });
+
+//                        AccessController.doPrivileged(new PrivilegedAction() {
```
Comment (pumudu88) [inline modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQConnection.java]: commented codes ?

File: modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQConnectionDelegate_8_0.java [hunk 3]
```diff
@@ -116,6 +116,15 @@ public ProtocolVersion makeBrokerConnection(BrokerDetails brokerDetail) throws A
             settings.setTrustStoreCertType(sslConfig.getCertType());
         }
 
+        if(settings.getKeyStorePath() == null && settings.getKeyStorePassword() ==null && settings
+                .getTrustStorePath() == null && settings.getTrustStorePassword() == null){
+            if (_logger.isDebugEnabled())
+            {
+                _logger.debug("Loading keystore and truststore from system properties");
```
Comment (a5anka) [inline modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQConnectionDelegate_8_0.java]: I think this information helps a lot in debugging ssl issues.
Therefore shall we make the log level to WARN?

File: modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQConnectionFactory.java [hunk 4]
```diff
@@ -274,11 +276,30 @@ public Connection createConnection() throws JMSException
...
+//        AccessController.doPrivileged(new PrivilegedAction() {
+//            public Object run() {
+//                if(!removeVersion91.get()) {
+//                    System.setProperty(ClientProperties.AMQP_VERSION, "0-91");
+//                } else {
+//                    System.clearProperty(ClientProperties.AMQP_VERSION);
+//                }
+//                return null; // nothing to return
+//            }
+//        });
```
Comment (a5anka) [inline modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQConnectionFactory.java]: Shall we get rid of the commented code?

File: modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQConnectionFactory.java [hunk 5]
```diff
@@ -289,11 +310,29 @@ public Connection createConnection() throws JMSException
...
+//        AccessController.doPrivileged(new PrivilegedAction() {
+//            public Object run() {
+//                if(!removeBURL.get()) {
+//                    System.setProperty("qpid.dest_syntax" , "BURL");
+//                } else {
+//                    System.clearProperty("qpid.dest_syntax");
+//                }
+//                return null; // nothing to return
+//            }
+//        });
```
Comment (a5anka) [inline modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQConnectionFactory.java]: Shall we get rid of the commented code?

File: modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQConnection.java (hunk 2 above)
Comment (pumudu88) [inline modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQConnection.java]: it's good if we can point this to a public jira.

**Comment by a5anka** [comment]
Shall we "git squash" these two commits together?

**Pull request closed without merging**